packages/patternfly-4/react-core/src/components/Select/CheckboxSelectOption.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Select/select';
+import { default as checkStyles } from '@patternfly/react-styles/css/components/Check/check';
+import { css } from '@patternfly/react-styles';
+import { SelectConsumer, KeyTypes } from './selectConstants';
+
+export interface CheckboxSelectOptionProps {
+  /** The value for the option */
+  children?: string;
+  /** Additional classes added to the Select Option */
+  className?: string;
+  /** Internal index of the option */
+  index?: number;
+  /** The value for the option */
+  value: string;
+  /** Flag indicating if the option is disabled */
+  isDisabled?: boolean;
+  /** Internal flag indicating if the option is checked */
+  isChecked?: boolean;
+  /** Internal callback for ref tracking */
+  sendRef?: (ref: React.ReactNode, index: number) => void;
+  /** Internal callback for keyboard navigation */
+  keyHandler?: (index: number, position: string) => void;
+  /** Optional callback for click event */
+  onClick?: (event: React.MouseEvent | React.ChangeEvent) => void;
+}
+
+export class CheckboxSelectOption extends React.Component<CheckboxSelectOptionProps> {
```

Comment:
Does this need to be a class if there is no state, @redallen? Not sure if there is an alternative to the componentDidMount and componentDidUpdate functions in this case. Should they be modifying a state?

Reply:
The several components mentioned all require lifecycle methods to work properly, so I'm unsure if they can be converted to functional components opposed to classes.

Reply:
@kmcfaul is correct.